evap/contributor/tests/test_views.py

```diff
@@ -292,6 +293,25 @@ def test_display_request_buttons(self):
         self.assertEqual(page.body.decode().count("Request changes"), 0)
         self.assertEqual(page.body.decode().count("Request creation of new account"), 2)
 
+    def test_contributor_evaluation_edit_main_language(self):
+        self.evaluation.main_language = "en"
+        self.evaluation.save()
+
+        response = self.app.get(self.url, user=self.editor, status=200)
+        form = response.forms["evaluation-form"]
+
+        with self.assertRaises(ValueError):
+            form["main_language"] = "x"
+
+        for lang, __ in settings.LANGUAGES:
+            form["main_language"] = lang
+            form.submit(name="operation", value="save")
+            self.assertEqual(Evaluation.objects.get().main_language, lang)
+
+        with self.assertRaises(ValueError):
+            form["main_language"] = "some_other_wrong_value"
+            form.submit(name="operation", value="save")
```

Comment:
Like the "Invalid value doesn't work"-block in 303, I think this will already fail on setting the form field as django webtest throws when we try to set a value that is not in the choices list. I think testing that server-side validation works is a good idea here, not sure if it can be done with WebTest form framework though. I think in the past we always manually crafted POST requests to achieve this.